### ISSUE TITLE
how to compute TL-ID and TG-ID?

### ISSUE THREAD

**Issue by laolongboy** [rotemtzaban/STIT#12]
Thanks.

**Comment by yaseryacoob** [comment]
There is something odd about the similarity metrics. The values can exceed 1.0. When I calculated them on source and edited videos using another StyleGAN editing approach, I got such values (1.04 and 1.09) for Local and global. I looked at your code, and I think it is all sensible since depending on head motion in the real videos, identity can be "better" controlled in the edit. It makes me wonder how useful these two metrics are in their current form. Even the reduced form of comparing identity within each video alone (i.e., source and edited) doesn't necessarily say much, the edited can have higher id preservation.

Thanks for sharing!

**Comment by yaseryacoob** [comment]
@rotemtzaban 
I completely agree that the ID metrics by in large are delivering what we actually want. So I don't mean to be critical of TL/TG-ID, I was intrigued when I saw the table in the paper, and on first look it didn't occur to me that the metric can stray above 1.0.  I am not sure that aiming for 1.0 solves the puzzle of how to compare the IDs, especially in your paper that the edits include age and gender. It makes me wonder if ID is even a relevant metric for these edits.  Even with face expression edits, ID may get erratic as it should, since the actual face is different, despite us mentally calling it the same ID. 

Add to it that the ID networks by and large are, in my opinion, not suitable for the objective. I don't think they capture the subtleties of the problem, but it is all what we have.

Anyway, to avoid a philosophical discussion, I wonder whether replacing the divide operation by cross correlation is a safer choice for at least face expression edits? Alternatively, the "ideal" criteria for performance can be posed as (1) the variance in identity over the entire edited video is zero (2) the closest real-frame ID distance to a given edited frame is 1.0 (for non age/gender examples), i.e., the edited frame is indistinguishable from a real one. I wish I can come up with better metrics, the editing task is likely ill-defined.

Cheers

**Comment by rmokady** [comment]
Hi @yaseryacoob,
TL/TG-ID metrics were deliberately designed to measure temporal coherence and *not* the quality of identity preservation in the individual frame edits / inversions themselves.

Your proposed minimal variance metric can also target the same goal, but please note that it suffers from the same flaw and actually doesn't let you identify cases where the editing method destroys part of the original content. For example, consider the case of a network that outputs the exact same image for all frames. This network will score perfectly on the minimal variation metric. This may be indented, as such a video is indeed perfectly consistent, but it wouldn't be as clear that there's an over-smoothing effect.

In other words - synthesized videos can absolutely reach better temporal coherence than real ones, and our metric does capture this. But it also gives you a way to identify this scenario if it's not desirable (which it won't be in most cases).

Your concern is of course justified. Temporal coherence alone is definitely not enough, we might want to better measure the *overall realism* of videos, including not only temporal consistency but also a per-frame realism and maybe even ensuring that the motion in the video makes sense. To my knowledge - using a single metric for all of those components is still an open question.
